src/dmangle.d

```diff
@@ -833,8 +833,13 @@ extern (C++) const(char)* mangleExact(FuncDeclaration fd)
 
 extern (C++) void mangleToBuffer(Type t, OutBuffer* buf)
 {
-    scope Mangler v = new Mangler(buf);
-    v.visitWithMask(t, 0);
+    if (t.deco)
+        buf.writestring(t.deco);
```

Comment:
`visitWithMask` already contains this caching logic. See:
https://github.com/dlang/dmd/blob/f2dc24d56f347c6640147aa14b0b6e1e278f667b/src/dmangle.d#L178